sklearn_numba_dpex/kmeans/tests/test_kmeans.py

```diff
@@ -411,3 +417,39 @@ def test_kmeans_plusplus_dataorder():
     )
 
     assert_allclose(centers_c, centers_fortran)
+
+
+def test_error_raised_on_invalid_group_sizes():
+    n_samples = 10
+    n_features = 2
+    n_clusters = 2
+    sub_group_size = 64
+    work_group_size = 500  # invalid because is not a multiple of sub_group_size
+    dtype = np.float32
+
+    with pytest.raises(ValueError):
```

Comment:
Could you please check the content of the error message as well?

```python
    expected_msg = re.escape(
        "..."  # TODO
    )
    with pytest.raises(ValueError, match=expected_msg):
        ...
```

In addition to checking the correctness of the message it will make the tests easier to read.